clientv3/concurrency/session.go

```diff
@@ -107,3 +114,14 @@ func WithTTL(ttl int) SessionOption {
 		}
 	}
 }
+
+// WithContext assigns a context to the session instead of defaulting to
+// using the client context. This is useful for canceling NewSession and
+// Close operations without having to close the client. If the context is
```

Comment:
canceling NewSession and Close operations `immediately`? (Close operation now has a upper bound waiting time)